buildSrc/src/main/java/org/opensearch/gradle/zipplugin/ZipPublish.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.gradle.zipplugin;
+
+import java.util.*;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.publish.PublishingExtension;
+import org.gradle.api.publish.maven.MavenPublication;
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin;
+import java.nio.file.Path;
+import org.gradle.api.Task;
+
+public class ZipPublish implements Plugin<Project> {
+    private Project project;
+
+    public final static String EXTENSION_NAME = "zipmavensettings";
+    public final static String PUBLICATION_NAME = "mavenzip";
+    public final static String STAGING_REPO = "zipstaging";
+    public final static String MAVEN_ZIP_PUBLISH_POM_TASK = "generatePomFileFor"
+        + ZipPublishUtil.capitalize(PUBLICATION_NAME)
+        + "Publication";
+    public final static String LOCALMAVEN = "publishToMavenLocal";
+    public final static String LOCAL_STAGING_REPO_PATH = "/build/local-staging-repo";
+    public String zipDistributionLocation = "/build/distributions/";
+
+    private void configMaven(Project project) {
+        final Path buildDirectory = project.getRootDir().toPath();
+        project.getPluginManager().apply(MavenPublishPlugin.class);
+        project.getExtensions().configure(PublishingExtension.class, publishing -> {
+            publishing.repositories(repositories -> {
+                repositories.maven(maven -> {
+                    maven.setName(STAGING_REPO);
+                    maven.setUrl(buildDirectory.toString() + LOCAL_STAGING_REPO_PATH);
+                });
+            });
+            publishing.publications(publications -> {
+                publications.create(PUBLICATION_NAME, MavenPublication.class, mavenZip -> {
+                    ZipPublishExtension extset = project.getExtensions().findByType(ZipPublishExtension.class);
+                    // Getting the Zip group from created extension
+                    String zipGroup = extset.getZipGroup();
+                    String zipArtifact = project.getName();
+                    String zipVersion = ZipPublishUtil.getProperty("version", project);
```

Comment:
The `zipVersion` property is required. Get rid of `ZipPublishUtil.getProperty` and just do `project.property()`. 

If you want a default, I believe you can do `project.findProperty('sampleNew') ?: 'default value'`.

Reply:
https://github.com/opensearch-project/OpenSearch/pull/2988#discussion_r856592640
If thats the case I will remove the Util's, but having Util's will definitely ease to add more methods and reduces the modification of the core logic and will also reduce some hard codings.